src/Common/tests/Tests/System/Collections/Generic/ArrayBuilderTests.cs

```diff
@@ -0,0 +1,365 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Linq;
+using Xunit;
+
+namespace System.Collections.Generic.Tests
+{
+    public interface IGenerator<out T>
+    {
+        T Generate(int seed);
+    }
+
+    public abstract class ArrayBuilderTests<T, TGenerator> where TGenerator : IGenerator<T>, new()
```

Comment:
I was having a very hard time with xUnit since it's very to hard to generate non-default Ts for use in MemberData-- see xunit/xunit#993 which I raised the other day. (tl;dr You can't have an abstract method which you call from the MemberData methods, for example, since they have to be static.)

A bright idea that I had was to have a `TGenerator` parameter we could call `Activator.CreateInstance` on which would enscapulate the abstract methods.